src/pages/Settings/ChangePwModal.tsx

```diff
@@ -1,142 +1,155 @@
+    reset,
     formState: { errors, isValid },
   } = useForm<IFormInputs>({
     mode: "onChange",
   });
 
-  return createPortal(
-    <ModalDialog close={onClose}>
-      <h3 className="font-bold">{t("settings.change_pw_a")}</h3>
-      <form
-        className="my-5 flex flex-col items-center justify-center text-center"
-        onSubmit={handleSubmit(changePwHandler)}
-      >
-        <article className="w-full py-2 md:w-10/12">
-          <InputField
-            {...register("oldPassword", {
-              required: t("setup.password_error_empty"),
-              pattern: {
-                value: /^[a-zA-Z0-9.-]*$/,
-                message: t("setup.password_error_chars"),
-              },
-              minLength: {
-                value: 8,
-                message: t("setup.password_error_length"),
-              },
-              onChange: onChangeOldPw,
-            })}
-            type="password"
-            value={oldPassword}
-            label={t("settings.old_pw")}
-            errorMessage={errors.oldPassword}
-            {...keyHandlers}
-          />
-        </article>
-        <article className="w-full py-2 md:w-10/12">
-          <InputField
-            {...register("newPassword", {
-              required: t("setup.password_error_empty"),
-              pattern: {
-                value: /^[a-zA-Z0-9.-]*$/,
-                message: t("setup.password_error_chars"),
-              },
-              minLength: {
-                value: 8,
-                message: t("setup.password_error_length"),
-              },
-              onChange: onChangeNewPw,
-            })}
-            type="password"
-            value={newPassword}
-            label={t("settings.new_pw")}
-            errorMessage={errors.newPassword}
-            {...keyHandlers}
-          />
-          {isCapsLockOn && <CapsLockWarning />}
-        </article>
-        <article className="flex w-full flex-col justify-around gap-6 pt-8 text-white md:w-2/3 xl:flex-row">
-          <button
-            className="bd-button-red flex items-center justify-center px-2"
-            onClick={onClose}
-            type="button"
-          >
-            <XMarkIcon className="inline h-6 w-6" />
-            <span className="p-2">{t("settings.cancel")}</span>
-          </button>
-          <ButtonWithSpinner
-            type="submit"
-            className="bd-button flex items-center justify-center px-2"
-            disabled={!isValid}
-            loading={isLoading}
-            icon={<ArrowPathIcon className="inline h-6 w-6" />}
-          >
-            <span className="p-2">{t("settings.change_pw_a")}</span>
-          </ButtonWithSpinner>
-        </article>
-      </form>
-    </ModalDialog>,
-    MODAL_ROOT,
+  return (
+    <>
+      <ConfirmModal
+        disclosure={confirmModal}
+        headline={t("settings.change_pw_a")}
+        isLoading={isLoading}
+        isFormModal={
+          <form onSubmit={handleSubmit(changePwHandler)}>
```

Comment:
I tried to define this part as a component further up but when validation kicks in it re-renders and fucks up the focus within the modal. Not sure how to solve this in a simple way. Tried a simple memo, but that didn't help.

Reply:
Lets keep it that way then.